backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,666 @@
+
+$pre-values = {
+                “ghj1jfg2”,
+                “dsa4cxz5”,
+                13,
+                9999,
+                5019
+            }
+
+INSERT INTO userAnswer (hash, password, variant, cid, moment) VALUES($pre-values[0], $pre-values[1], $pre-values[2], $pre-values[3], $pre-values[4])
+
+//Get the selected user by hash
+SELECT password,timesSubmitted,timesAccessed,grade from userAnswer WHERE hash=$hash;
+
+***
+send parameters input to showDuggaservice.php
+***
+send 
+{
+    $opt = “SAVDU”
+    $hash = “ghj1jfg2”
+}
+
+***
+Gather service output
+***
+Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using aid
+***
+
+	SQL-query: DELETE FROM userAnswer ORDER BY aid DESC LIMIT 1;
+			DELETE FROM listentries WHERE cid = 9999;
+			DELETE FROM course WHERE cid = 9999;
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #5 - 279 - 287
+update submitted dugga
+
+***
+Pre-req: !isset($grade) || !$grade > 1 &&
+isset($dbpwd) && strcmp($hashpwd,$dbpwd) === 0
+***
+
+***
+Pre-req variables
+***
+Pre-querys:
+$preValuesCourse = {	9999
+			1
+};
+
+INSERT INTO course(cid, creator) VALUES ($preValuesCourse[0], $preValuesCourse[1]);
+
+$preValuesListentries = {   9999, 
+                            "Inserttobedeleted",
+                            "UNK",
+                            4,
+                            12,
+                            2,
+                            1,
+                            1337,
+                            5019,
+                            1,
+                            1,
+                            0, 
+                            "UNK" }; 
+
+SQL-query:  INSERT INTO listentries(cid,entryname,link,kind,pos,creator,visible,vers,moment,gradesystem,highscoremode,feedbackenabled,feedbackquestion) 
+            VALUES ($preValuesListentries[0], $preValuesListentries[1], $preValuesListentries[2], $preValuesListentries[3], $preValuesListentries[4], $preValuesListentries[5], $preValuesListentries[6], $preValuesListentries[7], $preValuesListentries[8], $preValuesListentries[9], $preValuesListentries[10],$preValuesListentries[11],$preValuesListentries[12]);
+
+$pre-values =
+{
+    9999,
+    “dfg4zxc5”,
+    “asdfasdf”,
+    5019
+};
+SQL-query: INSERT INTO userAnswer(cid,hash,password,moment) VALUES ($pre-values[0], $pre-values[1], $pre-values[2], $pre-values[3]);
```

Comment:
i presume the same should apply in this insert, does moment have the correct value?

Reply:
moment is now removed and searched for instead, to make sure the right moment is used for each test.